flink-runtime/src/test/java/org/apache/flink/runtime/io/disk/FileChannelManagerImplTest.java

```diff
@@ -0,0 +1,192 @@
+					!fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX)) {
+				fail("The file channel manager test process does not create target directories in time, " +
+						"its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			Process kill = Runtime.getRuntime().exec("kill " + fileChannelManagerTestProcess.getProcessId());
+			kill.waitFor();
+			assertEquals("Failed to send SIG_TERM to process", 0, kill.exitValue());
+
+			deadline = Deadline.now().plus(TEST_TIMEOUT);
+			while (fileChannelManagerTestProcess.isAlive() && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (fileChannelManagerTestProcess.isAlive()) {
+				fail("The file channel manager test process does not terminate in time, its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			// Checks if the directories are cleared.
+			assertThat("The file channel manager test process does not remove the tmp shuffle directories after termination, " +
+							"its output is \n" + fileChannelManagerTestProcess.getProcessOutput(),
+					fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX),
+					is(false));
+		} finally {
+			if (fileChannelManagerTestProcess != null) {
+				fileChannelManagerTestProcess.destroy();
+			}
+		}
+	}
+
+	private boolean fileOrDirExists(File rootTmpDir, String namePattern) {
+		File[] candidates = rootTmpDir.listFiles((dir, name) -> name.contains(namePattern));
+		return candidates != null && candidates.length > 0;
+	}
+
+	/**
+	 * The {@link FileChannelManagerCleanupRunner} instance running in a separate JVM process.
+	 */
+	private static class FileChannelManagerTestProcess extends TestJvmProcess {
+		private final boolean callerHasHook;
+		private final String tmpDirectories;
+		private final String couldKillSignalFilePath;
+
+		public FileChannelManagerTestProcess(boolean callerHasHook, String tmpDirectories, String couldKillSignalFilePath) throws Exception {
+			this.callerHasHook = callerHasHook;
+			this.tmpDirectories = tmpDirectories;
+			this.couldKillSignalFilePath = couldKillSignalFilePath;
+		}
+
+		@Override
+		public String getName() {
+			return "File Channel Manager Test";
+		}
+
+		@Override
+		public String[] getJvmArgs() {
+			return new String[]{
+					Boolean.toString(callerHasHook),
+					tmpDirectories,
+					couldKillSignalFilePath
+			};
+		}
+
+		@Override
+		public String getEntryPointClassName() {
+			return FileChannelManagerCleanupRunner.class.getName();
+		}
+	}
+
+	/**
+	 * The entry point class to test the file channel manager cleanup with shutdown hook.
+	 */
+	public static class FileChannelManagerCleanupRunner {
+
+		public static void main(String[] args) throws Exception{
+			boolean callerHasHook = Boolean.parseBoolean(args[0]);
+			String tmpDirectory = args[1];
+			String couldKillSignalFilePath = args[2];
+
+			FileChannelManager manager = new FileChannelManagerImpl(new String[]{tmpDirectory}, DIR_NAME_PREFIX);
+
+			if (callerHasHook) {
```

Comment:
It is better to give some comment explanations here why we need `callerHasHook` for testing. E.g. we want to verify the case that both the upper component and its internal `FileChannelManagerImpl` have the shutdown hook in practice like `IOManager`.

Reply:
Agree with that and have modified the code accordingly.